### ISSUE TITLE
Test failure: test_web3_commands_via_attached_console

### ISSUE THREAD

**Issue by lithp** [ethereum/trinity#1213]
This one happens all the time. Example failure: https://circleci.com/gh/ethereum/trinity/129067

Another failure: https://circleci.com/gh/ethereum/trinity/128660

**Comment by lithp** [comment]
https://circleci.com/gh/ethereum/trinity/129067 has an interesting cause. I think `trinity` starts downloading headers too quickly! The test assumes that `w3.eth.getBlock('latest')` will return the genesis. However, if the trinity node is too quick then it will have already downloaded some headers, and the test will fail because it waits forever, looking for the genesis hash.

**Comment by cburgdorf** [comment]
Ha, that's an easy fix! We may just ask for `w3.eth.getBlock(0)` (or is it `0x0`?)  instead.

**Comment by lithp** [comment]
After I fixed that problem I ran into a few more problems. My current theory is that:
- Something is happening which takes up a LOT of cpu, so our 40 second timeout isn't long enough. Bumping the timeout to 60 seconds made the test pass more often.
- `AsyncProcessRunner` attaches `stderr` and `stdout` to pipes, and the test stops reading from the pipes as soon as `contains_all` returns. My theory is that the pipes fill up, then the trinity process hangs and stops sending messages to the `trinity attach` process. I added a coro which continually reads from stdout/stderr but `attached_trinity.expect()` blocks the event loop.
